### PR TITLE
MAGN-8072 Color Range Node for Flood Editor

### PR DESCRIPTION
**Pull request by marimano**
### Purpose

Move some computation logic from view to model, because it is needed in wpf-less Reach

### Reviewers

@pboyer

### CODE REVIEW

File: src/Libraries/CoreNodes/Color.cs [hunk 1]
```diff
@@ -171,20 +171,17 @@ public static Color Lerp(Color start, Color end, double t)
...
+            num[0] = (int)Math.Round((start.Alpha * d1)    + (end.Alpha * d2));
+            num[1] = (int)Math.Round((start.Red * d1)      + (end.Red * d2));
+            num[2] = (int)Math.Round((start.Green * d1)    + (end.Green * d2));
+            num[3] = (int)Math.Round((start.Blue * d1)     + (end.Blue * d2));
 
-            return ByARGB((int)(num[0] / (d1 + d2)),
-                (int)(num[1] / (d1 + d2)),
-                (int)(num[2] / (d1 + d2)),
-                (int)(num[3] / (d1 + d2)));
+            return ByARGB(num[0], num[1], num[2], num[3]);
```
Comment (marimano) [inline src/Libraries/CoreNodes/Color.cs]: Just a simplification here
dsc_1341

**Comment by pboyer** [comment]
@ikeough Can you review this? I'm not really familiar with this part of the code.

File: src/Libraries/CoreNodes/Color.cs [hunk 2]
```diff
@@ -171,20 +171,17 @@ public static Color Lerp(Color start, Color end, double t)
             }
 
             // Calculate the weighted average
-            var num = new double[4];
+            var num = new int[4];
 
-            var d1 = 1 - Math.Sqrt(Math.Pow(t - 0.0, 2));
-            var d2 = 1 - Math.Sqrt(Math.Pow(t - 1.0, 2));
+            var d1 = 1 - t;
```
Comment (ikeough) [inline src/Libraries/CoreNodes/Color.cs]: Ha! Thanks @marimano.

**Comment by ikeough** [comment]
@marimano LGTM. Thanks!

**Comment by ikeough** [comment]
@marimano @pboyer I've merged this. Sorry I forgot to do so earlier.

**Pull request merged**